lib/entities.js

```diff
@@ -0,0 +1,231 @@
+  };
+  const $set = {'meta.updated': Date.now()};
+  const $inc = {'entity.batchInvalidationCount': 1};
+  const collection = database.collections['tokenization-entity'];
+  const result = await collection.updateOne(
+    query, {$set, $inc}, database.writeOptions);
+
+  // return `true` if the update occurred
+  return result.result.n !== 0;
+}
+
+/**
+ * Removes an entity record identified by `internalId`.
+ *
+ * @param {object} options - Options to use.
+ * @param {Buffer} options.internalId - The internal ID for the entity.
+ *
+ * @returns {Promise<object>} The entity record.
+ */
+export async function _remove({internalId} = {}) {
+  const collection = database.collections['tokenization-entity'];
+  const query = {'entity.internalId': internalId};
+  await collection.deleteOne(query, database.writeOptions);
+}
+
+/**
+ * Sets the `batchId` associated with an open token batch for the entity
+ * identified by `internalId`. The open token batch can be either pinned
+ * or unpinned. A pinned token batch is one where the minimum identity
+ * assurance for that batch does not change. An unpinned token batch is
+ * the opposite: the minimum identity assurance for the batch can be
+ * changed -- these token batches inherit the `minAssuranceForResolution` that
+ * is set on the entity itself. Setting `minAssuranceForResolution` to `-1` or
+ * using its default value will cause the batch ID to be associated with an
+ * unpinned token batch.
+ *
+ * @param {object} options - Options to use.
+ * @param {Buffer} options.internalId - The internal ID for the entity.
+ * @param {Buffer} options.batchId - The open token batch ID for the entity.
+ * @param {number} [options.minAssuranceForResolution=-1] - Minimum level of
+ *   identity assurance required for token resolution for the given batch; for
+ *   an unpinned batch omit this value or set it to `-1`.
+ * @param {Buffer} [options.oldBatchId] - The old open token batch ID for the
+ *   entity; pass this to only update the open batch ID if there is a match.
+ *
+ * @returns {Promise<boolean>} Returns true if update occurred.
+ */
+export async function _setOpenTokenBatchId({
+  internalId, batchId, oldBatchId, minAssuranceForResolution = -1
+} = {}) {
+  const query = {'entity.internalId': internalId};
+  if(oldBatchId !== undefined) {
+    query[`entity.openBatch.${minAssuranceForResolution}`] = oldBatchId;
+  }
+  const $set = {
+    'meta.updated': Date.now()
+  };
+  $set[`entity.openBatch.${minAssuranceForResolution}`] = batchId;
+
+  const collection = database.collections['tokenization-entity'];
+  const result = await collection.updateOne(
+    query, {$set}, database.writeOptions);
+
+  // return `true` if the update occurred
+  return result.result.n !== 0;
+}
+
+/**
+ * Upserts a new entity. If the entity already exists, its time to live will be
+ * updated, but its `minAssuranceForResolution` will not be changed. To change
+ * the entity's `minAssuranceForResolution`
```

Comment:
It looks like this comment is unfinished.

Reply:
Thanks! Fixed in c4336da90e23b8c91bd395a755a3bfa42299b1d3.